probe-rs/src/probe/jlink/mod.rs

```diff
@@ -535,29 +598,26 @@ impl DebugProbe for JLink {
                     // Send the reset sequence (> 50 0-bits).
                     iter::repeat(true).take(64)
                     // Send the JTAG to SWD sequence.
-                    .chain(jtag_to_swd_sequence.iter().copied())
-                    // Send the reset sequence again in case we were in SWD mode already (> 50 0-bits).
-                    .chain(iter::repeat(true).take(64))
-                    // Send 10 idle line bits.
-                    .chain(iter::repeat(false).take(10));
+                    .chain(jtag_to_swd_sequence.iter().copied());
 
                 // Construct the direction sequence for reset sequence.
                 let direction =
                     // Send the reset sequence (> 50 0-bits).
                     iter::repeat(true).take(64)
                     // Send the JTAG to SWD sequence.
-                    .chain(iter::repeat(true).take(16))
-                    // Send the reset sequence again in case we were in SWD mode already (> 50 0-bits).
-                    .chain(iter::repeat(true).take(64))
-                    // Send 10 idle line bits.
-                    .chain(iter::repeat(true).take(10));
+                    .chain(iter::repeat(true).take(16));
 
                 // Send the init sequence.
                 // We don't actually care about the response here.
                 // A read on the DPIDR will finalize the init procedure and tell us if it worked.
                 jlink.swd_io(direction, swd_io_sequence)?;
+
+                // Perform a line reset
+                self.swd_line_reset()?;
                 log::debug!("Sucessfully switched to SWD");
 
+                // enable sticky overrun
```

Comment:
Good catch, enabling the sticky overrun behaviour is now done in the `ArmCommunicationInterface`, because we are configuring the debug port there already.